spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/RabbitConnectionFactoryBean.java

```diff
@@ -375,6 +375,15 @@ public void setExceptionHandler(ExceptionHandler exceptionHandler) {
 		this.connectionFactory.setExceptionHandler(exceptionHandler);
 	}
 
+	/**
+	 * Add custom properties to the connection factory's clientProperties.
+	 * @param properties the properties to add.
+	 * @since 1.5.6
+	 */
+	public void setCustomClientProperties(Map<String, Object> properties) {
+		this.connectionFactory.getClientProperties().putAll(properties);
+	}
```

Comment:
Doh - mistake I didn't see the existing method - however, that is bad (bug) because it replaces properties set up by the client itself; we need a getter to allow that; will fix to add.